include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -137,6 +146,24 @@ namespace internal
               Number *                                      hessians_quad,
               Number *                                      scratch_data,
               const bool add_into_values_array);
+
+    static Eval
+    create_evaluator_tensor_product(
+      const internal::MatrixFreeFunctions::UnivariateShapeData<Number>
+        *  univariate_shape_data,
+      bool request_even_odd)
+    {
+      Eval eval(request_even_odd ? univariate_shape_data->shape_values_eo :
+                                   univariate_shape_data->shape_values,
```

Comment:
Do I miss anything or would it be possible to compute `request_even_odd` from here, i.e., see a variable
```c++
  const bool request_even_odd = variant == evaluate_evenodd;
```
This would allow you to skip the second argument in all places below and hence make the code more compact.

Reply:
Good catch, thanks and done!